htdocs/core/class/commonobject.class.php

```diff
@@ -9472,6 +9472,12 @@ public function deleteCommon(User $user, $notrigger = false, $forcechilddeletion
 			}
 		}
 
+		// Delete linked object
+		$res = $this->deleteObjectLinked();
```

Comment:
This block should be before the deleteExteafields because order must be the oposite if create. Create should be
Insert object
Insert extrafields
Insert links
Insert other 
Call trigger

So delete should be
Call trigger
Delete other
Delete links (the part you added)
Delete extrafields
Delete object

So just move just before the delete of extrafields...